src/evd/classic/evd.cpp

```diff
@@ -18,19 +19,11 @@ void evd_classic(struct matrix_t Data_matr, struct matrix_t Eigen_vectors, struc
 
     int is_not_diagonal = 0;
 
-    // Build the auxiliary matrices
-
-    double *P, *temp;
-    P = (double*) malloc(sizeof(double) * m * m);
-    temp = (double*) malloc(sizeof(double) * m * m);
-
```

Comment:
I think we should decide on whether we want to change the input matrix A or how we should handle this. As Esref has said it is important for testing. We should talk about this today.